sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs

```diff
@@ -0,0 +1,191 @@
+            CompareModels(model, model2, format);
+        }
+
+        private bool AssertFailures(RoundTripStrategy<T> strategy, string format, string serviceResponse, ModelReaderWriterOptions options)
+        {
+            bool result = false;
+            if (IsXmlWireFormat && (strategy.IsExplicitJsonRead || strategy.IsExplicitJsonWrite) && format == "W")
+            {
+                if (strategy.IsExplicitJsonRead)
+                {
+                    if (strategy.GetType().Name.StartsWith("ModelJsonConverterStrategy"))
+                    {
+                        //we never get to the interface implementation because JsonSerializer errors before that
+                        Assert.Throws<JsonException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                        result = true;
+                    }
+                    else
+                    {
+                        Assert.Throws<InvalidOperationException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                        result = true;
+                    }
+                }
+
+                if (strategy.IsExplicitJsonWrite)
+                {
+                    Assert.Throws<InvalidOperationException>(() => { var data = strategy.Write(ModelInstance, options); });
+                    result = true;
+                }
+            }
+            else if (ModelInstance is not IJsonModel<T> && format == "J")
+            {
+                Assert.Throws<FormatException>(() => { T model = (T)strategy.Read(serviceResponse, ModelInstance, options); });
+                Assert.Throws<FormatException>(() => { var data = strategy.Write(ModelInstance, options); });
+                result = true;
+            }
+            return result;
+        }
+
+        internal static Dictionary<string, BinaryData> GetRawData(object model)
+        {
+            Type modelType = model.GetType();
+            while (modelType.BaseType != typeof(object) && modelType.BaseType != typeof(ValueType))
+            {
+                modelType = modelType.BaseType!;
+            }
+            var propertyInfo = modelType.GetField("_rawData", BindingFlags.Instance | BindingFlags.NonPublic);
+            return propertyInfo?.GetValue(model) as Dictionary<string, BinaryData> ?? throw new InvalidOperationException($"unable to get raw data from {model.GetType().Name}");
+        }
+
+        [Test]
+        public void ThrowsIfUnknownFormat()
+        {
+            ModelReaderWriterOptions options = new ModelReaderWriterOptions("x");
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Write(ModelInstance, options));
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Read<T>(new BinaryData("x"), options));
+
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Write((IPersistableModel<object>)ModelInstance, options));
```

Comment:
Please remove the cast to ensure the method works without it.

Reply:
Without the cast it will automatically use the `T` version which we do 3 lines up.  This forces the call into the non-Generic overload.